### PR TITLE
[459547] add close action to Xtext Builder Console

### PR DESCRIPTION
**Pull request by cdietrich**
[459547] add close action to Xtext Builder Console

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=459547

**Comment by cdietrich** [comment]
please review

File: org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/debug/XtextBuildConsolePageParticipant.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import org.eclipse.ui.part.IPageBookViewPage;
+
+/**
+ * @author Christian Dietrich - Initial contribution and API
+ * 
+ * {@link IConsolePageParticipant} that adds a close action to Xtext Build Console
+ * 
+ * @since 2.12
+ */
+public class XtextBuildConsolePageParticipant implements IConsolePageParticipant {
```
Comment (cdietrich) [inline org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/debug/XtextBuildConsolePageParticipant.java]: just did it the same way as jdt does

Comment (kthoms) [inline org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/debug/XtextBuildConsolePageParticipant.java]: Ok

File: org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/debug/XtextBuildConsolePageParticipant.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		return null;
+	}
+
+	@Override
+	public void init(IPageBookViewPage page, IConsole console) {
+		fCloseAction = new CloseConsoleAction(console);
+        
+        IToolBarManager manager = page.getSite().getActionBars().getToolBarManager();
+        manager.appendToGroup(IConsoleConstants.LAUNCH_GROUP, fCloseAction);
+
```
Comment (cdietrich) [inline org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/debug/XtextBuildConsolePageParticipant.java]: done

File: org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/debug/XtextBuildConsolePageParticipant.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	@Override
+	public Object getAdapter(@SuppressWarnings("rawtypes") Class adapter) {
+		return null;
+	}
+
+	@Override
+	public void init(IPageBookViewPage page, IConsole console) {
+		fCloseAction = new CloseConsoleAction(console);
+
```
Comment (cdietrich) [inline org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/debug/XtextBuildConsolePageParticipant.java]: done

**Pull request merged**